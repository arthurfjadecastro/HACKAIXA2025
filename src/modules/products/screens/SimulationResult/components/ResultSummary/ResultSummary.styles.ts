import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  resultCard: {
    backgroundColor: '#FFFFFF',
    borderRadius: 12,
    padding: 32,
    marginHorizontal: 8,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 4,
  },

  successTitleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 24,
  },

  successIcon: {
    marginRight: 8,
  },

  successTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#004AAD',
  },

  summarySection: {
    marginBottom: 16,
  },

  sectionTitle: {
    fontSize: 18,
    fontWeight: '600',
    color: '#333333',
    marginBottom: 16,
  },

  summaryRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: 8,
    borderBottomWidth: 1,
    borderBottomColor: '#F0F0F0',
  },

  label: {
    fontSize: 14,
    color: '#757575',
  },

  value: {
    fontSize: 14,
    fontWeight: '600',
    color: '#333333',
  },

  highlightValue: {
    fontSize: 16,
    color: '#004AAD',
  },
});
