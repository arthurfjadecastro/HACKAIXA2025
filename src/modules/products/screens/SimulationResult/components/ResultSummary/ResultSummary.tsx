import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { Text } from '@/design-system/components';
import { AmortizationToggle } from '../AmortizationToggle';
import { LoanSchedule } from '../../types';

interface ResultSummaryProps {
  amount: string;
  months: number;
  result: {
    rate?: number;
    rateAnnual?: number;
    amortizationType?: 'PRICE' | 'SAC';
  } | null | undefined;
  loanSchedule: LoanSchedule;
  isHabitacao: boolean;
  amortizationType: 'PRICE' | 'SAC';
  onAmortizationToggle: (type: 'PRICE' | 'SAC') => void;
  formatCurrency: (value: number) => string;
}

export const ResultSummary: React.FC<ResultSummaryProps> = ({
  amount,
  months,
  result,
  loanSchedule,
  isHabitacao,
  amortizationType,
  onAmortizationToggle,
  formatCurrency,
}) => {
  return (
    <View style={styles.resultCard}>
      <View style={styles.successTitleContainer}>
        <Ionicons name="checkmark-circle" size={24} color="#22C55E" style={styles.successIcon} />
        <Text style={styles.successTitle}>Simulação Concluída!</Text>
      </View>
      
      <View style={styles.summarySection}>
        <Text style={styles.sectionTitle}>Resumo</Text>
        
        <View style={styles.summaryRow}>
          <Text style={styles.label}>Valor do empréstimo:</Text>
          <Text style={styles.value}>R$ {amount}</Text>
        </View>
        
        <View style={styles.summaryRow}>
          <Text style={styles.label}>Parcelas:</Text>
          <Text style={styles.value}>
            {isHabitacao && amortizationType === 'SAC' 
              ? `${months}x`
              : `${months}x de ${formatCurrency(loanSchedule.monthlyInstallment)}`
            }
          </Text>
        </View>
        
        <View style={styles.summaryRow}>
          <Text style={styles.label}>Sistema de amortização:</Text>
          <Text style={styles.value}>
            {isHabitacao ? amortizationType : (result?.amortizationType || 'Price')}
          </Text>
        </View>
        
        {/* Toggle PRICE/SAC apenas para habitação */}
        {isHabitacao && (
          <AmortizationToggle
            amortizationType={amortizationType}
            onToggle={onAmortizationToggle}
          />
        )}
        
        <View style={styles.summaryRow}>
          <Text style={styles.label}>Valor da parcela:</Text>
          <Text style={StyleSheet.flatten([styles.value, styles.highlightValue])}>
            {isHabitacao && amortizationType === 'SAC' 
              ? 'Parcela Decrescente'
              : formatCurrency(loanSchedule.monthlyInstallment)
            }
          </Text>
        </View>
        
        <View style={styles.summaryRow}>
          <Text style={styles.label}>Total a pagar:</Text>
          <Text style={styles.value}>
            {formatCurrency(loanSchedule.totalWithInterest)}
          </Text>
        </View>
        
        <View style={styles.summaryRow}>
          <Text style={styles.label}>Total de juros:</Text>
          <Text style={styles.value}>
            {formatCurrency(loanSchedule.totalInterest)}
          </Text>
        </View>
        
        <View style={styles.summaryRow}>
          <Text style={styles.label}>Taxa de juros:</Text>
          <Text style={styles.value}>
            {result?.rate ? (result.rate).toFixed(2) : '2.00'}% a.m. ({result?.rateAnnual ? result.rateAnnual.toFixed(1) : ((Math.pow(1 + (result?.rate || 2)/100, 12) - 1) * 100).toFixed(1)}% a.a. efetiva)
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  resultCard: {
    backgroundColor: '#FFFFFF',
    borderRadius: 12,
    padding: 20, // Reduzido de 32 para 20
    marginHorizontal: 8,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 4,
  },

  successTitleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 16, // Reduzido de 24 para 16
  },

  successIcon: {
    marginRight: 8,
  },

  successTitle: {
    fontSize: 18, // Reduzido de 20 para 18
    fontWeight: 'bold',
    color: '#004AAD',
  },

  summarySection: {
    marginBottom: 12, // Reduzido de 16 para 12
  },

  sectionTitle: {
    fontSize: 16, // Reduzido de 18 para 16
    fontWeight: '600',
    color: '#333333',
    marginBottom: 12, // Reduzido de 16 para 12
  },

  summaryRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: 6, // Reduzido de 8 para 6
    borderBottomWidth: 1,
    borderBottomColor: '#F0F0F0',
  },

  label: {
    fontSize: 13, // Reduzido de 14 para 13
    color: '#757575',
  },

  value: {
    fontSize: 13, // Reduzido de 14 para 13
    fontWeight: '600',
    color: '#333333',
  },

  highlightValue: {
    fontSize: 14, // Reduzido de 16 para 14
    color: '#004AAD',
  },
});
