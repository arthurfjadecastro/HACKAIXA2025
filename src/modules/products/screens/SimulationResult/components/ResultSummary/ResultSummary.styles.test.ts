import { styles } from './ResultSummary.styles';

describe('ResultSummary.styles', () => {
  it('should export styles object', () => {
    expect(styles).toBeDefined();
    expect(typeof styles).toBe('object');
  });

  it('should have resultCard style properties', () => {
    expect(styles.resultCard).toBeDefined();
    expect(styles.resultCard.backgroundColor).toBe('#FFFFFF');
    expect(styles.resultCard.borderRadius).toBe(12);
  });

  it('should have successTitleContainer style properties', () => {
    expect(styles.successTitleContainer).toBeDefined();
    expect(styles.successTitleContainer.flexDirection).toBe('row');
  });

  it('should have consistent style structure', () => {
    expect(styles).toMatchSnapshot();
  });
});
