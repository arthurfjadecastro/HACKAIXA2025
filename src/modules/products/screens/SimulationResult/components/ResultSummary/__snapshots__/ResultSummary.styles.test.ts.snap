// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultSummary.styles should have consistent style structure 1`] = `
{
  "highlightValue": {
    "color": "#004AAD",
    "fontSize": 16,
  },
  "label": {
    "color": "#757575",
    "fontSize": 14,
  },
  "resultCard": {
    "backgroundColor": "#FFFFFF",
    "borderRadius": 12,
    "elevation": 4,
    "marginHorizontal": 8,
    "padding": 32,
    "shadowColor": "#000",
    "shadowOffset": {
      "height": 2,
      "width": 0,
    },
    "shadowOpacity": 0.1,
    "shadowRadius": 8,
  },
  "sectionTitle": {
    "color": "#333333",
    "fontSize": 18,
    "fontWeight": "600",
    "marginBottom": 16,
  },
  "successIcon": {
    "marginRight": 8,
  },
  "successTitle": {
    "color": "#004AAD",
    "fontSize": 20,
    "fontWeight": "bold",
  },
  "successTitleContainer": {
    "alignItems": "center",
    "flexDirection": "row",
    "justifyContent": "center",
    "marginBottom": 24,
  },
  "summaryRow": {
    "alignItems": "center",
    "borderBottomColor": "#F0F0F0",
    "borderBottomWidth": 1,
    "flexDirection": "row",
    "justifyContent": "space-between",
    "paddingVertical": 8,
  },
  "summarySection": {
    "marginBottom": 16,
  },
  "value": {
    "color": "#333333",
    "fontSize": 14,
    "fontWeight": "600",
  },
}
`;
